packages/dds/tree/src/events/events.ts

```diff
@@ -128,16 +129,16 @@ export class EventEmitter<E extends Events<E>> implements IEventEmitter<E> {
         };
     }
```

Comment:
Unfortunately you can't do it that way, unless I'm misunderstanding. You need the `as` either way, which is why I opted to let the return type be inferred for brevity.

`Property 'emit' is protected in type 'EventEmitter<E>' but public in type '{ emit: <K extends keyof Events<E>>(eventName: K, ...args: Parameters<E[K]>) => void; }'.`

Reply:
I guess this would also require making emit public, which I think makes more sense than violating scoping. I can imagine classes that want a public emitter anyway (ex: a class that is not exported, but has helpers that want to send events).

I'd suggest just making emit public on this class, and if you don't like that, banning the subclassing ussage pattern and just not exporting this class. Along with moving the create function outside the class, that would work cleanly

Reply:
I'd still prefer the create function at the module level so we can safely package export it publicly and make a separate choice about if we want to stabilize the class (possibly not package export it, or mark it as internal, or export it at a later time once we are more confident it the api stability).

Remember that classes have issues in public APIs. There is an undocument rule that subclassers of this don't name members "off" or "listeners" for example: those are both names someone might pick accidentally.

Even within this repo we have had ceases of accidental member name collisions (experimental shared tree overwrote the shared object logger for example) and it a much worse issue externally, so I'd like to aim for not package exporting classes at all.

Even if we decide to export it eventually, I'd like to have a way to package export thew create function sooner than that.

Reply:
Okay, I understand. Sounds good to me.